packages/p2p/src/stores/buy-sell-store.js

```diff
@@ -167,10 +168,20 @@ export default class BuySellStore extends BaseStore {
                       contact_info: values.contact_info,
                   }
                 : {}),
-        });
+        };
+        if (values.rate !== null) {
+            payload.rate = values.rate;
+        }
+
+        const order = await requestWS({ ...payload });
 
         if (order.error) {
             this.form_props.setErrorMessage(order.error.message);
+            this.setShouldShowPopup(false);
+            // TODO: Will remove this once https://github.com/binary-com/deriv-app/pull/5141 PR is merged
+            setTimeout(() => {
+                this.setShowRateChangePopup(true);
+            }, 250);
```

Comment:
Why do we need a setTimeout to show the popup?

Reply:
This is to prevent a glitch that occurs due to animation when a modal is closed and the subsequent modal is opened. Will remove this once https://github.com/binary-com/deriv-app/pull/5141 PR is merged